### PR TITLE
Updated to arrow 0.9.0

### PR DESCRIPTION
**Pull request by i-walker**
I hope I did not break anything :) 

I had problems with the task wrapper:
//task wrapper(type: Wrapper) {
  //  gradleVersion = "5.1.2"
//} 

It always through an error if I uncomment it.

### CODE REVIEW

**Comment by pakoito** [comment]
What error do you see?

File: build.gradle [hunk 1]
```diff
@@ -1,6 +1,6 @@
 buildscript {
-    ext.kotlin_version = '1.3.10'
-    ext.arrow_version = '0.8.2-SNAPSHOT'
+    ext.kotlin_version = '1.3.21'
+    ext.arrow_version = '0.9.0-SNAPSHOT'
```
Comment (pakoito) [inline build.gradle:3]: 0.9.0 is stable now :D

File: build.gradle [hunk 2]
```diff
@@ -37,30 +37,31 @@ repositories {
 dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
     compile "io.arrow-kt:arrow-ank:$arrow_version"
-    compile "io.arrow-kt:arrow-core:$arrow_version"
+    compile("io.arrow-kt:arrow-core:$arrow_version")
```
Comment (pakoito) [inline build.gradle]: it's called arrow-core-data now. If you've added it below this becomes unnecessary!

File: build.gradle [hunk 3]
```diff
@@ -37,30 +37,31 @@ repositories {
...
+    compile("io.arrow-kt:arrow-core-extensions:$arrow_version")
+    compile("io.arrow-kt:arrow-extras-extensions:$arrow_version")
+    kapt("io.arrow-kt:arrow-annotations-processor:$arrow_version")
+    kapt("io.arrow-kt:arrow-meta:$arrow_version")
 
-    compile "io.arrow-kt:arrow-free:$arrow_version" //optional
-    compile "io.arrow-kt:arrow-instances-free:$arrow_version" //optional
+    compile("io.arrow-kt:arrow-free:$arrow_version") //optional
+    compile("io.arrow-kt:arrow-free-extensions:$arrow_version") //optional
     compile "io.arrow-kt:arrow-mtl:$arrow_version" //optional
```
Comment (pakoito) [inline build.gradle]: Add the parens here for consistency too.

File: build.gradle (hunk 1 above)
Comment (i-walker) [reply build.gradle:3]: I know, but this is the only way to call the dependencies. This was in the past an issue, which I also resolved on the webpage. Maybe, I have the wrong maven URL, but then we should change it also on the webpage. https://arrow-kt.io/docs/

File: build.gradle (hunk 2 above)
Comment (i-walker) [reply build.gradle]: thanks

File: build.gradle (hunk 1 above)
Comment (i-walker) [reply build.gradle:3]: > What error do you see?

Screenshot 2019-03-15 at 00 11 38

should we rename it ? I am not an expert in gradle :)

**Comment by pakoito** [comment]
Remove the wrapper task, and -SNAPSHOT from the Arrow version.

**Comment by i-walker** [comment]
> - Remove the wrapper task <- Done ! 
> - and -SNAPSHOT from the Arrow version <- This won't build if you check out the source You will see there is no 0.9.0 Version defined https://oss.jfrog.org/artifactory/oss-snapshot-local/io/arrow-kt/arrow-free/

If there is another source can you send it to me ?

**Comment by i-walker** [comment]
As suspected: 

A problem occurred configuring root project 'Category-Theory-for-Programmers.kt'.
> Could not resolve all files for configuration ':runtimeClasspath'.
   > Could not resolve io.arrow-kt:arrow-free:0.9.0.
     Required by:
         project :
      > Could not resolve io.arrow-kt:arrow-free:0.9.0.
         > Could not get resource 'https://oss.jfrog.org/artifactory/oss-snapshot-local/io/arrow-kt/arrow-free/0.9.0/arrow-

**Comment by 1Jajen1** [comment]
It should be free-data right?
Shows here: https://github.com/arrow-kt/arrow/blob/0.9.0/README.md#next-development-version-090

**Comment by i-walker** [comment]
> It should be free-data right?
> Shows here: https://github.com/arrow-kt/arrow/blob/0.9.0/README.md#next-development-version-090

no thats not the issue [Check this](https://oss.jfrog.org/artifactory/oss-snapshot-local/io/arrow-kt/arrow-free-data/)

**Comment by i-walker** [comment]
But I found the real problem. If you execute ./gradlew runAnk there are some failures within the ank files. If I may, I will correct them. I hope the owners won't mind. Then the build should be successful. 
## Stacktrace
Ank Compile: [5] 1.9-function-types.md | Used Heap: 117.0 MiB
[✗ /Users/ulti_mate/IdeaProjects/Category-Theory-for-Programmers.kt/build/site/1.9-function-types.md [7]
Exception in thread "main" 

```

val f: (Either<Int, Double>) -> String = {
  when (it) {
    is Left -> if (it.a < 0) "Negative int" else "Positive int"
    is Right -> if (it.b < 0.0) "Negative double" else "Positive double"
  }
}
```
error: unresolved reference: Either
val f: (Either<Int, Double>) -> String = {
        ^
error: 'when' expression must be exhaustive, add necessary 'else' branch
  when (it) {
  ^
error: unresolved reference: Left
    is Left -> if (it.a < 0) "Negative int" else "Positive int"
       ^
error: unresolved reference: Right
    is Right -> if (it.b < 0.0) "Negative double" else "Positive double"
       ^


> Task :runAnk FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runAnk'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 27s
2 actionable tasks: 1 executed, 1 up-to-date

**Comment by 1Jajen1** [comment]
Hmm that is definitly something wrong with arrows packages :/ sucks. The ank errors are expected, some modules changed iirc so the imports need to be redone on some of them.

**Comment by i-walker** [comment]
Done 💪🏾

File: local.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (pakoito) [inline local.properties]: As it says :D

Comment (i-walker) [reply local.properties]: I am not sure if that worked out, where can I see the whole file tree of a PR in GitHub

**Comment by raulraja** [comment]
thanks @i-walker !

**Pull request merged**